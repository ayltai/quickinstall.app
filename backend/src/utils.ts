import { ErrorReporting, } from '@google-cloud/error-reporting';

const errors = new ErrorReporting();

export const handleError = (error : any) => {
    if (process.env.NODE_ENV === 'production') {
        errors.report(error);
    } else {
        console.error(error);
    }
};
