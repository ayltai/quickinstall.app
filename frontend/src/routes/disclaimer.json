{
    "metadata" : {
        "tags" : [ ]
    },
    "sys"      : {
        "space"       : {
            "sys" : {
                "type"     : "Link",
                "linkType" : "Space",
                "id"       : "rljsuvr9aw29"
            }
        },
        "id"          : "5JIZ4p8HrhBkuF5NI5F0dH",
        "type"        : "Entry",
        "createdAt"   : "2024-02-23T13:25:57.134Z",
        "updatedAt"   : "2024-02-23T13:25:57.134Z",
        "environment" : {
            "sys" : {
                "id"       : "development",
                "type"     : "Link",
                "linkType" : "Environment"
            }
        },
        "revision"    : 1,
        "contentType" : {
            "sys" : {
                "type"     : "Link",
                "linkType" : "ContentType",
                "id"       : "page"
            }
        },
        "locale"      : "en"
    },
    "fields"   : {
        "name"    : "Disclaimer",
        "content" : "# DISCLAIMER\n\n*Last updated February 14, 2024*\n\n## WEBSITE DISCLAIMER\n\nThe information provide by QuickInstall.app ('**we**', '**us**', or '**our**') on [https://quickinstall.app](https://quickinstall.app) (the 'Site') is for general information purposes only. All information on the Site is provided in good faith, however we make no representation or warranty of any kind, express or implied, regarding the accuracy, adequacy, validity, reliability, availability, or completeness of any information on the Site. UNDER NO CIRCUMSTANCE SHALL WE HAVE ANY LIABILITY TO YOU FOR ANY LOSS OR DAMAGE OF ANY KIND INCURRED AS A RESULT OF THE USE OF THE SITE OR RELIANCE ON ANY INFORMATION PROVIDED ON THE SITE. YOUR USE OF THE SITE AND YOUR RELIANCE ON ANY INFORMATION ON THE SITE IS SOLELY AT YOUR OWN RISK.\n\n## EXTERNAL LINKS DISCLAIMER\n\nThe Site may contain (or you may be sent through the Site) links to other websites or content belonging to or originating from third parties or links to websites and features. Such eternal links are not investigated, monitored, or checked for accuracy, adequacy, validity, reliability, availability, or completeness by us. WE DO NOT WARRANT, ENDORSE, GUARANTEE, OR ASSUME RESPONSIBILITY FOR THE ACCURACY OR RELIABILITY OF ANY INFORMATION OFFERED BY THIRD-PARTY WEBSITES LINKED THROUGH THE SITE OR ANY WEBSITE OR FEATURES. WE WILL NOT BE A PART TO OR IN ANY WAY BE RESPONSIBLE FOR MONITORING ANY TRANSACTION BETWEEN YOU AND THIRD-PARTY PROVIDERS OF PRODUCTS OR SERVICES.\n"
    }
}
