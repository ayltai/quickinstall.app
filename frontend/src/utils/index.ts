import * as Sentry from '@sentry/react';
import { FirebaseApp, } from 'firebase/app';
import { getAnalytics, logEvent, } from 'firebase/analytics';

export const handleError = (error : any) => {
    if (import.meta.env.PROD) {
        Sentry.captureException(error);
    } else {
        console.error(error);
    }
};

export const logPageView = (app : FirebaseApp, {
    pageName,
    route,
} : {
    pageName : string,
    route    : string,
}) => {
    try {
        logEvent(getAnalytics(app), 'page_view', {
            page_title    : pageName,
            page_location : route,
        });
    } catch (error) {
        handleError(error);
    }
};

export const logAddToCart = (app : FirebaseApp, {
    platform,
    itemId,
} : {
    platform : 'win' | 'mac',
    itemId   : string,
}) => {
    try {
        logEvent(getAnalytics(app), 'add_to_cart', {
            items : [
                {
                    item_id : `${platform}/${itemId}`,
                },
            ],
        });
    } catch (error) {
        handleError(error);
    }
};

export const logRemoveFromCart = (app : FirebaseApp, {
    platform,
    itemId,
} : {
    platform : 'win' | 'mac',
    itemId   : string,
}) => {
    try {
        logEvent(getAnalytics(app), 'remove_from_cart', {
            items : [
                {
                    item_id : `${platform}/${itemId}`,
                },
            ],
        });
    } catch (error) {
        handleError(error);
    }
};

export const logCheckout = (app : FirebaseApp, {
    platform,
    itemIds,
} : {
    platform : 'win' | 'mac',
    itemIds  : string[],
}) => {
    try {
        logEvent(getAnalytics(app), 'checkout', {
            items : itemIds.map((itemId : string) => ({
                item_id : `${platform}/${itemId}`,
            })),
        });
    } catch (error) {
        handleError(error);
    }
};

export const logSearch = (app : FirebaseApp, {
    platform,
    query,
} : {
    platform : 'win' | 'mac',
    query    : string,
}) => {
    try {
        logEvent(getAnalytics(app), 'search', {
            search_term : `${platform}/${query}`,
        });
    } catch (error) {
        handleError(error);
    }
};

export const logShare = (app : FirebaseApp, {
    product,
    itemId,
} : {
    product : 'win/packages' | 'mac/packages' | 'tweaks',
    itemId  : string,
}) => {
    try {
        logEvent(getAnalytics(app), 'share', {
            content_type : product,
            item_id      : itemId,
        });
    } catch (error) {
        handleError(error);
    }
};
