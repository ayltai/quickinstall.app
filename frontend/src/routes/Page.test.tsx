import React from 'react';
import createFetchMock from 'vitest-fetch-mock';

import { render, waitFor, } from '../utils/test';
import disclaimer from './disclaimer.json';
import { Page, } from './Page';

const fetch = createFetchMock(vi);
fetch.enableMocks();

vi.mock('react-router-dom', async () => ({
    ...await vi.importActual('react-router-dom'),
    useParams : () => ({
        page : 'about',
    }),
}));

describe('<Page />', () => {
    it('renders correctly', async () => {
        fetch.mockIf(request => request.url.startsWith('https://cdn.contentful.com'), request => {
            if (request.method === 'GET') return {
                status  : 200,
                body    : JSON.stringify(disclaimer),
                headers : {
                    'Content-Type' : 'application/json',
                },
            };

            return {
                status : 501,
            };
        });

        await waitFor(() => expect(render(<Page />)).toMatchSnapshot());
    });
});
