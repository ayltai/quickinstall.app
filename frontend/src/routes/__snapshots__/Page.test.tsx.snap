// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Page /> > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-173mpwd"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionFixed mui-fixed css-1jvst5f-MuiPaper-root-MuiAppBar-root"
          style="background-color: rgba(255, 255, 255, 0.32);"
        >
          <div
            class="MuiContainer-root MuiContainer-maxWidthLg css-kicy4c-MuiContainer-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
            >
              <a
                href="/"
              >
                <img
                  alt="app.name"
                  height="48"
                  src="/src/components/TopAppBar/SmallLogo.webp"
                  style="margin-right: 8px;"
                  width="48"
                />
              </a>
              <div
                class="MuiBox-root css-i9gxme"
              />
              <div
                class="MuiBox-root css-i9gxme"
              />
              <button
                aria-label="action.theme_toggle"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="LightModeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5M2 13h2c.55 0 1-.45 1-1s-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1m18 0h2c.55 0 1-.45 1-1s-.45-1-1-1h-2c-.55 0-1 .45-1 1s.45 1 1 1M11 2v2c0 .55.45 1 1 1s1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1m0 18v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1M5.99 4.58c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0s.39-1.03 0-1.41zm12.37 12.37c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0 .39-.39.39-1.03 0-1.41zm1.06-10.96c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0zM7.05 18.36c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </header>
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        />
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <div
            class="MuiBox-root css-jvmkpy"
          />
          <div
            aria-hidden="true"
            class="MuiBackdrop-root css-sox5kk-MuiBackdrop-root"
            style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
          >
            <div
              class="MuiStack-root css-6rg9iv-MuiStack-root"
            >
              <span
                class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-mod6k9-MuiCircularProgress-root"
                role="progressbar"
                style="width: 40px; height: 40px;"
              >
                <svg
                  class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                  viewBox="22 22 44 44"
                >
                  <circle
                    class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                    cx="44"
                    cy="44"
                    fill="none"
                    r="20.2"
                    stroke-width="3.6"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiStack-root css-b6lxw6-MuiStack-root"
        >
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <div
            class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
          >
            <div
              class="MuiStack-root css-3hgi49-MuiStack-root"
            >
              <div
                class="MuiStack-root css-nen11g-MuiStack-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-1ooch7a-MuiTypography-root"
                >
                  footer.header.product
                </span>
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1y0wc18-MuiTypography-root-MuiLink-root"
                  href="/#/pages/about"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-j4exjj-MuiTypography-root"
                  >
                    footer.product.about
                  </p>
                </a>
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1y0wc18-MuiTypography-root-MuiLink-root"
                  href="https://quickinstall.hashnode.dev"
                  rel="noreferrer"
                  target="_blank"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-j4exjj-MuiTypography-root"
                  >
                    footer.product.blog
                  </p>
                </a>
              </div>
              <div
                class="MuiStack-root css-nen11g-MuiStack-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-1ooch7a-MuiTypography-root"
                >
                  footer.header.support
                </span>
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1y0wc18-MuiTypography-root-MuiLink-root"
                  href="https://github.com/ayltai/quickinstall.app/issues/new?assignees=&labels=help+wanted&projects=&template=help_wanted.md&title="
                  rel="noreferrer"
                  target="_blank"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-j4exjj-MuiTypography-root"
                  >
                    footer.support.help
                  </p>
                </a>
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1y0wc18-MuiTypography-root-MuiLink-root"
                  href="https://github.com/ayltai/quickinstall.app/issues/new?assignees=&labels=bug&projects=&template=bug_report.md&title="
                  rel="noreferrer"
                  target="_blank"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-j4exjj-MuiTypography-root"
                  >
                    footer.support.bugs
                  </p>
                </a>
              </div>
              <div
                class="MuiStack-root css-nen11g-MuiStack-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-overline css-1ooch7a-MuiTypography-root"
                >
                  footer.header.terms
                </span>
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1y0wc18-MuiTypography-root-MuiLink-root"
                  href="/#/pages/terms"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-j4exjj-MuiTypography-root"
                  >
                    footer.terms.terms
                  </p>
                </a>
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1y0wc18-MuiTypography-root-MuiLink-root"
                  href="/#/pages/disclaimer"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-j4exjj-MuiTypography-root"
                  >
                    footer.terms.disclaimer
                  </p>
                </a>
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1y0wc18-MuiTypography-root-MuiLink-root"
                  href="/#/pages/privacy"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-j4exjj-MuiTypography-root"
                  >
                    footer.terms.privacy
                  </p>
                </a>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-yhebxb"
          >
            <div
              class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
            >
              <div
                class="MuiStack-root css-1jch9ch-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-4pw5dy-MuiTypography-root"
                >
                  footer.copyright
                </p>
                <div
                  class="MuiBox-root css-i9gxme"
                />
                <a
                  aria-label="footer.link.github"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                  href="https://github.com/ayltai/quickinstall.app"
                  tabindex="0"
                  target="_blank"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1srol0e-MuiSvgIcon-root"
                    data-testid="GitHubIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
                <a
                  aria-label="footer.link.linkedin"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                  href="https://linkedin.com/in/ayltai"
                  tabindex="0"
                  target="_blank"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1srol0e-MuiSvgIcon-root"
                    data-testid="LinkedInIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-173mpwd"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionFixed mui-fixed css-1jvst5f-MuiPaper-root-MuiAppBar-root"
        style="background-color: rgba(255, 255, 255, 0.32);"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-kicy4c-MuiContainer-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
          >
            <a
              href="/"
            >
              <img
                alt="app.name"
                height="48"
                src="/src/components/TopAppBar/SmallLogo.webp"
                style="margin-right: 8px;"
                width="48"
              />
            </a>
            <div
              class="MuiBox-root css-i9gxme"
            />
            <div
              class="MuiBox-root css-i9gxme"
            />
            <button
              aria-label="action.theme_toggle"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="LightModeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5M2 13h2c.55 0 1-.45 1-1s-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1m18 0h2c.55 0 1-.45 1-1s-.45-1-1-1h-2c-.55 0-1 .45-1 1s.45 1 1 1M11 2v2c0 .55.45 1 1 1s1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1m0 18v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1M5.99 4.58c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0s.39-1.03 0-1.41zm12.37 12.37c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0 .39-.39.39-1.03 0-1.41zm1.06-10.96c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0zM7.05 18.36c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </header>
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      />
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-jvmkpy"
        />
        <div
          aria-hidden="true"
          class="MuiBackdrop-root css-sox5kk-MuiBackdrop-root"
          style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            class="MuiStack-root css-6rg9iv-MuiStack-root"
          >
            <span
              class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-mod6k9-MuiCircularProgress-root"
              role="progressbar"
              style="width: 40px; height: 40px;"
            >
              <svg
                class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
      <div
        class="MuiStack-root css-b6lxw6-MuiStack-root"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <div
            class="MuiStack-root css-3hgi49-MuiStack-root"
          >
            <div
              class="MuiStack-root css-nen11g-MuiStack-root"
            >
              <span
                class="MuiTypography-root MuiTypography-overline css-1ooch7a-MuiTypography-root"
              >
                footer.header.product
              </span>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1y0wc18-MuiTypography-root-MuiLink-root"
                href="/#/pages/about"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-j4exjj-MuiTypography-root"
                >
                  footer.product.about
                </p>
              </a>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1y0wc18-MuiTypography-root-MuiLink-root"
                href="https://quickinstall.hashnode.dev"
                rel="noreferrer"
                target="_blank"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-j4exjj-MuiTypography-root"
                >
                  footer.product.blog
                </p>
              </a>
            </div>
            <div
              class="MuiStack-root css-nen11g-MuiStack-root"
            >
              <span
                class="MuiTypography-root MuiTypography-overline css-1ooch7a-MuiTypography-root"
              >
                footer.header.support
              </span>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1y0wc18-MuiTypography-root-MuiLink-root"
                href="https://github.com/ayltai/quickinstall.app/issues/new?assignees=&labels=help+wanted&projects=&template=help_wanted.md&title="
                rel="noreferrer"
                target="_blank"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-j4exjj-MuiTypography-root"
                >
                  footer.support.help
                </p>
              </a>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1y0wc18-MuiTypography-root-MuiLink-root"
                href="https://github.com/ayltai/quickinstall.app/issues/new?assignees=&labels=bug&projects=&template=bug_report.md&title="
                rel="noreferrer"
                target="_blank"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-j4exjj-MuiTypography-root"
                >
                  footer.support.bugs
                </p>
              </a>
            </div>
            <div
              class="MuiStack-root css-nen11g-MuiStack-root"
            >
              <span
                class="MuiTypography-root MuiTypography-overline css-1ooch7a-MuiTypography-root"
              >
                footer.header.terms
              </span>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1y0wc18-MuiTypography-root-MuiLink-root"
                href="/#/pages/terms"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-j4exjj-MuiTypography-root"
                >
                  footer.terms.terms
                </p>
              </a>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1y0wc18-MuiTypography-root-MuiLink-root"
                href="/#/pages/disclaimer"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-j4exjj-MuiTypography-root"
                >
                  footer.terms.disclaimer
                </p>
              </a>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1y0wc18-MuiTypography-root-MuiLink-root"
                href="/#/pages/privacy"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-j4exjj-MuiTypography-root"
                >
                  footer.terms.privacy
                </p>
              </a>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-yhebxb"
        >
          <div
            class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
          >
            <div
              class="MuiStack-root css-1jch9ch-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-4pw5dy-MuiTypography-root"
              >
                footer.copyright
              </p>
              <div
                class="MuiBox-root css-i9gxme"
              />
              <a
                aria-label="footer.link.github"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                href="https://github.com/ayltai/quickinstall.app"
                tabindex="0"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1srol0e-MuiSvgIcon-root"
                  data-testid="GitHubIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
              <a
                aria-label="footer.link.linkedin"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                href="https://linkedin.com/in/ayltai"
                tabindex="0"
                target="_blank"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1srol0e-MuiSvgIcon-root"
                  data-testid="LinkedInIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
